lsp-mode.el

```diff
@@ -5810,16 +5840,18 @@ an alist
       (cons name
             (lsp--imenu-create-hierarchical-index children?)))))
 
-(lsp-defun lsp--symbol-filter ((&SymbolInformation :location))
-  "Determine if SYM is for the current document."
-  ;; It's a SymbolInformation or DocumentSymbol, which is always in the current
-  ;; buffer file.
-  (when location
-    (not (eq (->> location
-                  (lsp:location-uri)
-                  (lsp--uri-to-path)
-                  (find-buffer-visiting))
-             (current-buffer)))))
+(lsp-defun lsp--symbol-filter ((&SymbolInformation :kind :location))
+  "Determine if SYM is for the current document and is to be shown."
+  ;; It's a SymbolInformation or DocumentSymbol, which is always in the
+  ;; current buffer file.
+  (or (and (not (null lsp-imenu-index-symbol-kinds))
```

Comment:
Shouldn't this be `and` instead of `or`?
If this is `or` how can it filter more than the old function?

Reply:
Also you can replace `(not (null a)) ` with just `a`

Reply:
> Shouldn't this be and instead of or?

No :)